### PR TITLE
Ihor/ray core/rich dropdown component

### PR DESCRIPTION
**Pull request by sick-sad-world**
Dropdown component for ray framework. When default <select> is not enough. Works as wrapper-enchancer for <select> component. Fallback as `.ray-select` if JS is not applied for some reason.
https://deploy-preview-252--ray-docs.netlify.com/

### CODE REVIEW

**Comment by i8ramin** [comment]
Looks like the icon is overlapping the drop down arrow 

5FC35C1C-A782-4AB3-80F9-881641D12ED7

File: docs/html/dropdown/dropdown--optgroups.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<div className="ray-dropdown">
```
Comment (adamraider) [inline docs/html/dropdown/dropdown--optgroups.html]: These .html files need to be valid html. These look like JSX, make sure you convert things like `className` to `class`, and `htmlFor` to `for`

File: docs/html/dropdown/dropdown--with-prepend.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        <option value="Pikachu">Pikachu</option>
+        <option value="Squirtle">Squirtle</option>
+        <option value="Charmander">Charmander</option>
+        <option value="Sonichu">Sonichu</option>
+      </select>
+      <label className="ray-dropdown__label" htmlFor="test-example">
+        Pokemon
+      </label>
+    </div>
+  </div>
```
Comment (adamraider) [inline docs/html/dropdown/dropdown--with-prepend.html]: Make sure all HTML is valid (not JSX) and please fix the indentation

File: packages/core/src/components/dropdown/_dropdown.scss [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        border: $ray-border-width solid $ray-color-gray-60;
+        border-top: 0;
+        border-radius: 0 0 $ray-border-radius $ray-border-radius;
+        max-height: $dropdown-max-height;
+        overflow-y: auto;
+        transition-duration: 0.25s;
+        transition-timing-function: ease-in;
+        transition-property: height, max-height;
+      }
+      &-list {
```
Comment (adamraider) [inline packages/core/src/components/dropdown/_dropdown.scss]: delete unused selector

File: packages/core/src/components/dropdown/index.js [hunk 4]
```diff
@@ -0,0 +1,353 @@
+import {
```
Comment (adamraider) [inline packages/core/src/components/dropdown/index.js]: Please write tests for this component. See other tests for examples.

This dropped our coverage by ~20%!

File: docs/html/dropdown/dropdown--optgroups.html (hunk 1 above)
Comment (sick-sad-world) [reply docs/html/dropdown/dropdown--optgroups.html]: thanks. copy-paste plays bad role

File: docs/html/dropdown/dropdown--with-prepend.html (hunk 2 above)
Comment (sick-sad-world) [reply docs/html/dropdown/dropdown--with-prepend.html]: done

File: packages/core/src/components/dropdown/_dropdown.scss (hunk 3 above)
Comment (sick-sad-world) [reply packages/core/src/components/dropdown/_dropdown.scss]: done

File: docs/html/dropdown/dropdown--optgroups.html (hunk 1 above)
Comment (sick-sad-world) [reply docs/html/dropdown/dropdown--optgroups.html]: done

File: packages/core/src/components/dropdown/index.js (hunk 4 above)
Comment (sick-sad-world) [reply packages/core/src/components/dropdown/index.js]: done

File: packages/core/src/components/dropdown/_dropdown.scss (hunk 3 above)
Comment (adamraider) [reply packages/core/src/components/dropdown/_dropdown.scss]: There's still Coverage decreased (-4.5%) , could you add more tests? You can view the coveralls report to see which lines need coverage

Comment (sick-sad-world) [reply packages/core/src/components/dropdown/_dropdown.scss]: added more tests

File: docs/components/dropdown.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+// destroy
+dropdown.destroy();
+
+// open
+dropdown.open();
+```
+
+| Method                     | Params               | Description                          |
+| -------------------------- | -------------------- | ------------------------------------ |
+| `Select.createAll`         | `HTMLElement:Object` | create all instances in document     |
```
Comment (adamraider) [inline docs/components/dropdown.md]: should be Dropdown.* here?

Comment (sick-sad-world) [reply docs/components/dropdown.md]: > Looks good, but no keyboard navigation support?

Yeah we split it out on primary and secondary features. Keyboard nav is secondary. We wan't to merge it to ray to unblock and finish GROW-9590. Btw custom language dropdown at footer of wework.com site which was used as prototype also doesn't have keyboard nav

**Pull request merged**